app/javascript/packages/pwned-passwords/cli.js

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env node
+
+import { parseArgs } from 'node:util';
+import { pipeline } from 'node:stream/promises';
+import { createWriteStream } from 'node:fs';
+import Progress from 'cli-progress';
+import Downloader from './downloader.js';
+
+const { values: flags } = parseArgs({
+  options: {
+    'range-start': { type: 'string' },
+    'range-end': { type: 'string' },
+    'max-size': { type: 'string', short: 'n' },
+    concurrency: { type: 'string' },
+    'out-file': { type: 'string', short: 'o' },
+  },
+});
+
+const {
+  'range-start': rangeStart,
+  'range-end': rangeEnd,
+  'max-size': maxSize,
+  concurrency,
+  'out-file': outFile,
+} = flags;
+
+const downloader = new Downloader({
+  rangeStart,
+  rangeEnd,
+  concurrency: concurrency ? Number(concurrency) : undefined,
+  maxSize: maxSize ? Number(maxSize) : undefined,
+});
+
+const outputStream = outFile ? createWriteStream(outFile) : process.stdout;
+
+if (outFile) {
+  const progressBar = new Progress.SingleBar({
+    format:
+      '[{bar}] {percentage}% | ETA {eta}s | {value}/{total} | {hashes} hashes (>= {hashMin} prevalence)',
+  });
+  downloader.once('start', ({ total }) => progressBar.start(total, 0, { hashes: 0, hashMin: 0 }));
+  downloader.on('download', () => progressBar.increment());
+  downloader.on('hashchange', ({ hashes, hashMin }) => progressBar.update({ hashes, hashMin }));
+  downloader.once('complete', () => progressBar.stop());
+}
+
+const result = await downloader.download();
+
+await pipeline(
```

Comment:
It's probably complex enough to deal with as a separate follow-on enhancement, but I think we could consider an idea mentioned by @zachmargolis to make this resumable.

The way I could imagine it working is...

1. When configured to write to a file, check if that file exists and has a number of lines less than the target size
2. If so, prompt to ask (or maybe support a flag to specify) if the download should pick up from where it left off
3. Automatically adjust `rangeStart` based on number of lines already present in the file, and initialize `commonHashes` with the current file data
   - TBD: How do we retain prevalence counts to know if any existing data should be ejected? 🤔 
5. Bind to [`SIGTERM`/`SIGINT` process events](https://nodejs.org/api/process.html#signal-events) and dump contents immediately to file if triggered